test/framework/src/test/java/org/elasticsearch/test/test/ESTestCaseTests.java

```diff
@@ -258,4 +265,38 @@ public void testSecureRandom() throws NoSuchAlgorithmException {
             }
         }
     }
+
+    public void testSecureRandomKeyGen() throws NoSuchAlgorithmException, NoSuchProviderException, ClassNotFoundException {
+        if (ESTestCase.inFipsJvm()) {
+            { // non-FIPS SecureRandom => FIPS KeyGenerator => Fails
+                final Exception e = expectThrows(Exception.class, () -> {
+                    final SecureRandom secureRandomNonFips = secureRandomNonFips();
+                    KeyGenerator keyGeneratorFips = KeyGenerator.getInstance("AES", "BCFIPS");
```

Comment:
do you need to set the BC provider here ? I would kinda expect that when in FIPS mode it was either first or the only provider.

Reply:
I removed it, and replaced it with BCFIPS provider name assertions in the tests.